include/dlaf/common/range2d.h

```diff
@@ -72,37 +72,41 @@ struct IteratorRange2D {
   }
 
   reference operator[](difference_type n) const noexcept {
-    return computeIndex2D(n);
+    auto it = *this;
+    it += n;
+    return *it;
   }
 
   IteratorRange2D& operator++() noexcept {
-    current_ = (++i_, computeIndex2D(i_));
+    current_ = computeIndex2D(++i_);
```

Comment:
Does `computeIndex2D` ever operate on anything but `i_` (after the changes)? Would it be clearer to change `computeIndex2D` to simply modify `current_` and use `i_`. I know in this case it's actually making things less explicit, but given that it's always used the same way, it might actually be less error-prone to have it modify the state? However, I think the current formulation is also good.

Reply:
It's a good point, it is not worth anymore. I fully agree with you that it is better to make the parameters and update value embedded in the method.

It's also true that the problem of the uncoupling between the `i_` and `current_` will still exist, but in this latter case any more complicated solution would look like an overshot to me.